adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -437,18 +454,22 @@ public OrderData authorisePayment(final HttpServletRequest request, final CartDa
                 getSessionService().setAttribute(SESSION_MD, paymentsResponse.getRedirect().getData().get(MD));
                 getSessionService().setAttribute(SESSION_PAYMENT_DATA, paymentsResponse.getPaymentData());
             }
+            if (KLARNA.equals(adyenPaymentMethod)) {
+                getSessionService().setAttribute(PAYMENT_METHOD, KLARNA);
+                getSessionService().setAttribute(SESSION_PAYMENT_DATA, paymentsResponse.getPaymentData());
```

Comment:
this is not connected to only Klarna this is needed for all paymentMethods.

Reply:
This value is being used only for klarna and all other payment methods are being identified by nulll value. If we add it for all payment methods, it will need to be updated for all other payment methods and will result in more if/else clauses.

Reply:
for now fine 👍